processor/simple_supply_tp/handler.py

```diff
@@ -97,6 +102,36 @@ def _create_record(state, public_key, payload):
         timestamp=payload.timestamp)
 
 
+def _transfer_record(state, public_key, payload):
+    if state.get_agent(payload.data.receiving_agent) is None:
+        raise InvalidTransaction('Agent with the public key {} does '
+                                 'not exist'.format(public_key))
+
+    record = state.get_record(payload.data.record_id)
+    if record is None:
+        raise InvalidTransaction('Record with the record id {} does not '
+                                 'exist'.format(payload.data.record_id))
+
+    if not _validate_record_owner(agent_id=public_key, record=record):
+        raise InvalidTransaction(
+            'Transaction signer is not the owner of the record')
+
+    state.transfer_record(
+        receiving_agent=payload.data.receiving_agent,
+        record_id=payload.data.record_id,
+        timestamp=payload.timestamp)
+
+
+def _validate_record_owner(agent_id, record):
+    latest_owner = ''
+    latest_timestamp = -1
+    for owner in record.owners:
```

Comment:
I think you should be able to do `latest_owner = max(record.owners, key: lambda x: x.timestamp).agent_id` instead of this explicit looping logic.